### PR TITLE
sci-visualization/gwyddion: Version bump to 2.47, EAPI bump 5 -> 6

### PR DESCRIPTION
**Pull request by g-braeunlich**
Package-Manager: portage-2.3.0

### CODE REVIEW

**Comment by g-braeunlich** [comment]
Tested without kde use flag due to conflicts with dependencies

File: sci-visualization/gwyddion/gwyddion-2.47.ebuild [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+	ruby? ( dev-ruby/narray )
+	sourceview? ( x11-libs/gtksourceview:2.0 )
+	xml? ( dev-libs/libxml2:2 )"
+
+DEPEND="${RDEPEND}
+	virtual/pkgconfig
+	doc? ( dev-util/gtk-doc )
+"
+
+REQUIRED_USE="${PYTHON_REQUIRED_USE}"
```
Comment (SoapGentoo) [inline sci-visualization/gwyddion/gwyddion-2.47.ebuild]: this needs to be `python?` conditional too

File: sci-visualization/gwyddion/gwyddion-2.47.ebuild [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+HOMEPAGE="http://gwyddion.net/"
+SRC_URI="http://gwyddion.net/download/${PV}/${P}.tar.xz"
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64 ~x86 ~amd64-linux ~x86-linux"
+IUSE="doc fits fftw gnome kde nls opengl perl python ruby sourceview xml X"
+
+RDEPEND="
+	media-libs/libpng:0
```
Comment (SoapGentoo) [inline sci-visualization/gwyddion/gwyddion-2.47.ebuild]: `:0=`

File: sci-visualization/gwyddion/gwyddion-2.47.ebuild [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+	x11-libs/cairo
+	x11-libs/gtk+:2
+	x11-libs/libXmu
+	x11-libs/pango
+	fits? ( sci-libs/cfitsio )
+	fftw? ( sci-libs/fftw:3.0 )
```
Comment (SoapGentoo) [inline sci-visualization/gwyddion/gwyddion-2.47.ebuild]: `:3.0=`

File: sci-visualization/gwyddion/gwyddion-2.47.ebuild [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+	gnome? ( gnome-base/gconf:2 )
+	kde? ( kde-frameworks/kdelibs:4 )
+	opengl? ( virtual/opengl x11-libs/gtkglext )
+	perl? ( dev-lang/perl )
```
Comment (SoapGentoo) [inline sci-visualization/gwyddion/gwyddion-2.47.ebuild]: does this package install perl modules?

File: sci-visualization/gwyddion/gwyddion-2.47.ebuild [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+	xml? ( dev-libs/libxml2:2 )"
+
+DEPEND="${RDEPEND}
+	virtual/pkgconfig
+	doc? ( dev-util/gtk-doc )
+"
+
+REQUIRED_USE="${PYTHON_REQUIRED_USE}"
+
+MAKEOPTS+=" V=1"
```
Comment (SoapGentoo) [inline sci-visualization/gwyddion/gwyddion-2.47.ebuild]: chuck this, rather do `emake -j1` in `src_compile` (have you looked whether the build system is fixable?)

File: sci-visualization/gwyddion/gwyddion-2.47.ebuild (hunk 4 above)
Comment (g-braeunlich) [inline sci-visualization/gwyddion/gwyddion-2.47.ebuild]: There is a "/usr/lib64/gwyddion/perl/Gwyddion/dump.pm"

File: sci-visualization/gwyddion/gwyddion-2.47.ebuild (hunk 5 above)
Comment (g-braeunlich) [inline sci-visualization/gwyddion/gwyddion-2.47.ebuild]: works for me with -j6

File: sci-visualization/gwyddion/gwyddion-2.47.ebuild (hunk 4 above)
Comment (SoapGentoo) [inline sci-visualization/gwyddion/gwyddion-2.47.ebuild]: so interfacing perl with normal build systems is messy. This is the wrong path (for which you're ofc not responsible). In general, packages installing `.pm`'s need to install them into Perl's `$VENDORLIB`, and that depends on Perl's subslot, i.e.:

1. You need to `:=` depend on perl
2. The `.pm`'s need to be installed correctly

Installing `.pm`'s correctly from non-Perl build systems is a slight nightmare at the moment, but @kentfredric has submitted a stack of helper functions to the ml to make this stuff half-way sane and in the spirit of `insinto`/`doins`. Once that stuff is merged, we can proceed here.

**Pull request closed without merging**